ChainedPICs requires 1 offset instead of 2.

rasheedmhd opened the pull request:
The current code requires two offsets when calling `ChainedPics::new`. 

The user of the crate is therefore required to provide 2 offsets when using this crate. 

Example:
`pub const OFFSET_1: u8 = 32;` and
`pub const OFFSET_2: u8 = OFFSET_1 + 8;` 

I believe this puts a little weight on the user which could be abstracted away in the implementation of the `ChainedPics::new`.

With `OFFSET_2`, if the user accidentally uses a number greater than 8 or less than 8, we end up with two undesirable cases
1. Leaving a gap between two PICs in the IDT
2. Overwriting some interrupts of the Primary PIC

By taking only one offset from the user, an offset of the Primary PIC we can compute the next offset inside the `ChainedPics::new` essentially avoiding the issue stated above.

rasheedmhd committed:
editted offsets

phil-opp commented:
Thanks for the PR! While it is a common choice to set `offset_2 = offset_1 + 8`, you are not required to do so. "Leaving a gap between two PICs in the IDT" might be even intended for some specific use cases. Given that this is a quite low-level crate, I think it's more important to stay as flexible and unopinionated as possible.

So I'm in favor of keeping the `new` method as-is. I'm open to adding a second constructor method to create a contiguous mapping though. For example, we could name it `new_contiguous`. This way, users can decide what they prefer and we don't break any existing code. What do you think?

rasheedmhd committed:
Added new_contiguous to ChainedPics

rasheedmhd committed:
Update Cargo.toml

rasheedmhd commented:
Thanks for attending to this PR. I totally agree with you. Thanks for pointing out that there are use cases that leaving a gap is the intended. It is something that I didn't know about and I just learnt from you. I am wish to know I such use case, just to learn more. I would be glad if you could point me in the right direction.

Taking your reply into consideration, I have reworked on my fork, retaining the old `new` implementation for backward compatibility and also added the` new_contiguous` implementation.

Closing this PR, to open a new PR for the new implementation. Thanks.
